### PR TITLE
Add user to correct self group

### PR DESCRIPTION
**Pull request by acron0**
Some other QoL changes

### CODE REVIEW

File: src/cljs/witan/ui/schema.cljs [hunk 1]
```diff
@@ -11,6 +11,14 @@
    :kixi.group/type   s/Keyword
    :kixi.group/id     uuid?})
 
+(def UserSchema
+  {:kixi.user/name   (s/maybe s/Str)
+   :kixi.user/id     (s/maybe s/Str)
```
Comment (Eleonore9) [inline src/cljs/witan/ui/schema.cljs:16]: Wouldn't this be a uuid?

File: src/cljs/witan/ui/schema.cljs [hunk 2]
```diff
@@ -11,6 +11,14 @@
...
+   :kixi.user/groups [(s/maybe s/Str)]
```
Comment (Eleonore9) [inline src/cljs/witan/ui/schema.cljs:17]: Wouldn't this be a vector of uuids?

File: src/cljs/witan/ui/schema.cljs [hunk 3]
```diff
@@ -11,6 +11,14 @@
...
+   :kixi.user/self-group (s/maybe s/Str)
```
Comment (Eleonore9) [inline src/cljs/witan/ui/schema.cljs:18]: Same here about str vs uuid

File: src/cljs/witan/ui/controllers/datastore.cljs [hunk 4]
```diff
@@ -194,6 +194,14 @@
     (.open js/window link)))
 
 (defmethod on-event
+  [:kixi.datastore.filestore/download-link-rejected "1.0.0"]
+  [{:keys [args]}]
+  (let [{:keys [kixi.comms.event/payload]} args
+        {:keys [reason]} payload]
+    (.alert js/window (str "Unable to download file: " (name reason)))
```
Comment (Eleonore9) [inline src/cljs/witan/ui/controllers/datastore.cljs:201]: Is `reason` the exception message as a string?

**Review by Eleonore9 — COMMENTED** [review]
Lgtm, just a few questions

File: src/cljs/witan/ui/schema.cljs (hunk 1 above)
Comment (acron0) [reply src/cljs/witan/ui/schema.cljs:16]: The transmission of UUIDs is a pain and in most cases we're only sending it back to the backend anyway, so no need to convert

File: src/cljs/witan/ui/schema.cljs (hunk 2 above)
Comment (acron0) [reply src/cljs/witan/ui/schema.cljs:17]: See above

File: src/cljs/witan/ui/schema.cljs (hunk 3 above)
Comment (acron0) [reply src/cljs/witan/ui/schema.cljs:18]: See above

File: src/cljs/witan/ui/controllers/datastore.cljs (hunk 4 above)
Comment (acron0) [reply src/cljs/witan/ui/controllers/datastore.cljs:201]: As a keyword, at the moment

File: src/cljs/witan/ui/schema.cljs (hunk 1 above)
Comment (Eleonore9) [reply src/cljs/witan/ui/schema.cljs:16]: ok, good to know!

File: src/cljs/witan/ui/controllers/datastore.cljs (hunk 4 above)
Comment (Eleonore9) [reply src/cljs/witan/ui/controllers/datastore.cljs:201]: ok

**Review by Eleonore9 — APPROVED** [review]
:shipit:

**Pull request merged**